tests/test_result_manager.py

```diff
@@ -25,9 +27,85 @@
 from model_analyzer.result.result_manager import ResultManager
 from model_analyzer.state.analyzer_state_manager import AnalyzerStateManager
 
+from filecmp import cmp
+from shutil import rmtree
+from unittest.mock import MagicMock, patch
+
 
 class TestResultManager(trc.TestResultCollector):
 
+    def setUp(self):
+        self._create_single_model_result_manager()
+        self._create_multi_model_result_manager()
+
+    def tearDown(self):
+        patch.stopall()
+
+    def test_single_model_csv_against_golden(self):
+        """
+        Match the csvs against the golden versions in
+        tests/common/single-model-ckpt
+        """
+        self._single_model_result_manager.tabulate_results()
+        self._single_model_result_manager.export_results()
+
+        self.assertTrue(
+            cmp(f"{ROOT_DIR}/single-model-ckpt/results/metrics-model-gpu.csv",
```

Comment:
have you locally tried to edit the csv to a different value to confirm that this will fail?

Reply:
No, but I have tested it the other way (modifying the new csv being outputted) and have observed the test case failing, so I have proof this is working as intended/